tests/test_rio_mask.py

```diff
@@ -6,17 +6,17 @@
 
 import affine
 import numpy as np
-from packaging.version import parse
 import pytest
 
 import rasterio
 from rasterio.crs import CRS
 from rasterio.rio.main import main_group
+from rasterio.env import GDALVersion
 
 
 # Custom markers.
 xfail_pixel_sensitive_gdal22 = pytest.mark.xfail(
```

Comment:
Should we be skipping instead of failing these tests?

Reply:
Yes. I've used xfail improperly in the past and this looks like another example.

Reply:
Agreed.  The [`pytest` docs](https://docs.pytest.org/en/latest/skipping.html) indicate this should be a skip.  Seeing a all those `x`'s in the `pytest` output suggests we have a bunch of unsolved bugs when in reality we just have tests requiring specific versions of GDAL.